### PR TITLE
Alert on asset download error

### LINKED ISSUE #39449: Refuse to execute script from because its MIME type ('') is not executable
### Describe the bug

Hey!  

We have a metabase instance deployed in a Azure Web App following the [documentation](https://www.metabase.com/docs/latest/installation-and-operation/running-metabase-on-azure).  It works really well except for some users who cannot access it because of this error:
`Refused to execute script from 'https://metabase.xxx.azurewebsites.net/app/dist/app-main.2557cfcbb410eb3199ba.js' because its MIME type ('') is not executable, and strict MIME type checking is enabled.` 
Most of the type, hard reload the page works but for some, they always got a white page because they get a 404 on the `app-main.xxx.js` script.

On this session, it loaded perfectly the first time, but I refreshed the page and got the problem:
image

### To Reproduce

1. Deploy Metabase on Azure Web App
2. Open the base URL

### Expected behavior

It should work all the time

### Logs

_No response_

### Information about your Metabase installation

```JSON
- Chrome lastest
- Azure Web app
- Container image: metabase/metabase:latest
```

### Severity

Blocking some users

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by npfitz**
Closes https://github.com/metabase/metabase/issues/39449

### Description
Adds an `onerror` handler to bundled scripts from webpack. If triggered, will display an alert to the user asking that they clear their cache and refresh the browser. Also logs a console error with some more developer related information. If there are multiple errors, we should only show 1 alert, but as many console errors as needed.

### How to verify

1. Clear your `release/frontend_client/app/dist folder`
2. Build a release version of mb with `yarn build-release`
3. Start the backend, then rename some of your built JS assets
4. open `localhost:3000`. You should see an alert pop up, and error messages in the JS console. Only 1 alert should display

### Demo
image
image

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: resources/frontend_client/index_template.html [hunk 1]
```diff
@@ -41,6 +41,7 @@
 
     <!-- If you modify this script, make sure you update the whitelisted Content-Security-Policy hash in metabase.server.middleware.security -->
     <script type="text/javascript">{{{bootstrapJS}}}</script>
+    <script type="text/javascript">{{{assetOnErrorJS}}}</script>
```
Comment (heypoom) [inline resources/frontend_client/index_template.html:44]: I wonder if we would need to modify `inline-js-hashes` there to add `frontend_client/inline_js/asset_loading_error.js` to this block?

```
    (mapv file-hash [ ;; inline script in index.html that sets `MetabaseBootstrap` and the like
                     "frontend_client/inline_js/index_bootstrap.js"
                     ;; inline script in init.html
                     "frontend_client/inline_js/init.js"])))
```

**Review by heypoom — APPROVED** [review]
Looks good to me! I'm not sure if we need to add the Content-Security-Policy hash for the inline js script though.

**Review by sloansparger — APPROVED** [review]

File: resources/frontend_client/inline_js/asset_loading_error.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+(function () {
+  window.mbErrorAlert = false;
+  window.AssetErrorLoad = function (tag) {
+    console.error(
+      `Could not download asset ${tag.src} from your metabase instance. 
+This shouldn't happen normally, but can happen in certain 
+instances where your browser has cached the index.html page, or 
+there are different versions of metabase behind the same load balancer. 
+If clearing your cache doesn't resolve the issue, please ensure your
+all your deployed instances of metabase are on the same version.`,
```
Comment (ranquild) [inline resources/frontend_client/inline_js/asset_loading_error.js:10]: `please ensure your all your` - typo?

File: resources/frontend_client/inline_js/asset_loading_error.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+(function () {
+  window.mbErrorAlert = false;
+  window.AssetErrorLoad = function (tag) {
```
Comment (ranquild) [inline resources/frontend_client/inline_js/asset_loading_error.js:3]: Maybe add this to our global `Metabase` variable? Not sure if it is set in production

File: resources/frontend_client/inline_js/asset_loading_error.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      `Could not download asset ${tag.src} from your metabase instance. 
+This shouldn't happen normally, but can happen in certain 
+instances where your browser has cached the index.html page, or 
+there are different versions of metabase behind the same load balancer. 
+If clearing your cache doesn't resolve the issue, please ensure your
+all your deployed instances of metabase are on the same version.`,
+    );
+
+    if (!window.mbErrorAlert) {
+      alert(
```
Comment (ranquild) [inline resources/frontend_client/inline_js/asset_loading_error.js:14]: I would probably not display this alert and only show the error message in console. I think we _might_ not understand some cases well enough and could show this message too often

**Review by ranquild — COMMENTED** [review]
Inline comments

File: resources/frontend_client/index_template.html (hunk 1 above)
Comment (ranquild) [reply resources/frontend_client/index_template.html:44]: Yes! @npfitz please check this is Safari as it's more strict about CSP.

**Review by ranquild — COMMENTED** [review]
Last comment about CSP

**Review by ranquild — APPROVED** [review]

**Pull request merged**